srml/contract/src/lib.rs

```diff
@@ -276,7 +279,7 @@ decl_module! {
 			//
 			// NOTE: this should go after the commit to the storage, since the storage changes
 			// can alter the balance of the caller.
-			gas::refund_unused_gas::<T>(&origin, gas_meter);
+			gas::refund_unused_gas::<T>(&origin, gas_meter)?;
```

Comment:
Note that here (and above) when we can't refund funds for some reason, we abort the execution of the extrinsic with an error, even though all changes are commited.

Moreover, we lose all calls recorded so far.